### PR TITLE
TCM-547 | Add dummy settings permissions screen

### PR DESCRIPTION
**Pull request by GorkemKarayel**
### Summary

Add dummy permissions to settings screen

### Issue

 [TCM-547](https://hyfnla.atlassian.net/browse/TCM-547)

### Video/Screenshot

permission_not_enabled

enable_permissions

settings

https://user-images.githubusercontent.com/17555866/123506469-08a8ed80-d66d-11eb-8d74-b7f36b51146b.mov

### CODE REVIEW

File: app/src/main/res/layout/row_permission.xml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    <com.trx.consumer.common.CommonLabel
+        android:id="@+id/lblPermissionName"
+        android:layout_width="wrap_content"
+        android:layout_height="30dp"
+        android:layout_marginStart="15dp"
+        android:layout_marginVertical="15dp"
+        android:gravity="center_vertical"
+        android:fontFamily="@font/atcarquette_regular"
+        android:textColor="#212121"
```
Comment (onodude) [inline app/src/main/res/layout/row_permission.xml:18]: sorry for the wrong specs, lets just use black

File: app/src/main/res/drawable/bg_permission_switch.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (onodude) [inline app/src/main/res/drawable/bg_permission_switch.xml:1]: would it be possible to set the attributes instead

File: app/src/main/res/drawable/bg_permission_switch_thumb.xml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (onodude) [inline app/src/main/res/drawable/bg_permission_switch_thumb.xml:1]: same here could we set an attribute for the switch and use default ui

**Review by onodude — CHANGES_REQUESTED** [review]
looking good @GorkemKarayel added a couple of questions

File: app/src/main/res/drawable/bg_permission_switch.xml (hunk 2 above)
Comment (GorkemKarayel) [reply app/src/main/res/drawable/bg_permission_switch.xml:1]: If I'm not wrong we cannot do it. The default switch view in Android is not as in the design. I had to add custom style to this reason. If you have suggestion I can fix it.

File: app/src/main/java/com/trx/consumer/models/common/PermissionModel.kt [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package com.trx.consumer.models.common
+
+class PermissionModel {
+
+    // Dummy property for permissions
+    var id: String = ""
```
Comment (VivekV95) [inline app/src/main/java/com/trx/consumer/models/common/PermissionModel.kt:6]: We try to use the primary constructor for properties whenever possible

File: app/src/main/java/com/trx/consumer/screens/permissions/PermissionViewHolder.kt [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+import androidx.appcompat.widget.SwitchCompat
+import com.trx.consumer.R
+import com.trx.consumer.common.CommonLabel
+import com.trx.consumer.common.CommonViewHolder
+import com.trx.consumer.models.common.PermissionModel
+
+class PermissionViewHolder(view: View) : CommonViewHolder(view) {
+
+    private val lblPermissionName: CommonLabel = view.findViewById(R.id.lblPermissionName)
+    private val scPermission: SwitchCompat = view.findViewById(R.id.scPermission)
```
Comment (VivekV95) [inline app/src/main/java/com/trx/consumer/screens/permissions/PermissionViewHolder.kt:13]: Looks like on Xpass we're going with the `switch` naming convention, so let's rename the property and view id to `switchPermission` to keep things consistent

File: app/src/main/res/drawable/bg_permission_switch.xml (hunk 2 above)
Comment (VivekV95) [reply app/src/main/res/drawable/bg_permission_switch.xml:1]: @onodude Do we need the switches to look exactly like figma? [This](https://user-images.githubusercontent.com/39810816/126553347-d56aa62a-b94f-4fb2-870e-27c60d974aba.png) is the default switch

File: app/src/main/res/drawable/bg_permission_switch_thumb.xml (hunk 3 above)
Comment (VivekV95) [reply app/src/main/res/drawable/bg_permission_switch_thumb.xml:1]: Same as [this](https://github.com/hyfn/trx-consumer-android/pull/151/files#r674304356)

**Review by VivekV95 — CHANGES_REQUESTED** [review]
One change and a question

File: app/src/main/java/com/trx/consumer/models/common/SettingsModel.kt [hunk 6]
```diff
@@ -21,6 +21,14 @@ class SettingsModel {
                     "$size active membership${if (size == 1) "" else "s"}"
                 }
                 SettingsType.EMAIL -> user?.email ?: "N/A"
+                SettingsType.PERMISSIONS -> {
```
Comment (devender197) [inline app/src/main/java/com/trx/consumer/models/common/SettingsModel.kt:24]: @onodude can we use [xpass](https://github.com/hyfn/xpass-android/blob/develop/app/src/main/java/com/xponential/xpass/screens/settings/SettingsModel.kt#L86) logic here. Except that we have parity issue here from [iOS](https://github.com/hyfn/trx-consumer-ios/pull/268/files#diff-dec0e355099c51081ed0e3a105c2df60167ec6098addbbceebbae234c432c861R55)

File: app/src/main/java/com/trx/consumer/models/common/SettingsModel.kt [hunk 7]
```diff
@@ -59,6 +68,7 @@ class SettingsModel {
                 add(create(null, SettingsType.LOGOUT))
                 if (BuildConfig.DEBUG) add(create(null, SettingsType.TEST_SCREENS))
                 if (BuildConfig.DEBUG) add(create(null, SettingsType.SHOW_MAINTENANCE))
+                if (BuildConfig.DEBUG) add(create(null, SettingsType.PERMISSIONS))
```
Comment (devender197) [inline app/src/main/java/com/trx/consumer/models/common/SettingsModel.kt:71]: Position of permission on setting screen is different than iOS [here](https://github.com/hyfn/trx-consumer-ios/pull/268/files#diff-65aea56938b0a6d5e75c4c8b3282908c82cc26805f590007a716fd65bbcf9e42R21)

File: app/src/main/res/layout/row_permission.xml [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+<com.trx.consumer.common.CommonView xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="60dp"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_marginBottom="15dp"
+    app:backgroundColor="@color/white">
+
+    <com.trx.consumer.common.CommonLabel
+        android:id="@+id/lblPermissionName"
```
Comment (devender197) [inline app/src/main/res/layout/row_permission.xml:11]: Please consider to name the view as lblTitle similar to iOS [here](https://github.com/hyfn/trx-consumer-ios/pull/268/files#diff-caf7fe970db6f71c85ff9f77344b63eaebcccd38a98556bcd5c81a4eb2d68d62R35)

File: app/src/main/res/layout/row_permission.xml [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+        android:fontFamily="@font/atcarquette_regular"
+        android:textColor="@color/black"
+        android:textSize="16sp"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        tools:text="Camera Permissions" />
+
+    <androidx.appcompat.widget.SwitchCompat
+        android:id="@+id/scPermission"
```
Comment (devender197) [inline app/src/main/res/layout/row_permission.xml:26]: Please consider to name view as `btnSelect` similar to iOS [here](https://github.com/hyfn/trx-consumer-ios/pull/268/files#diff-caf7fe970db6f71c85ff9f77344b63eaebcccd38a98556bcd5c81a4eb2d68d62R52)

File: app/src/main/res/layout/row_permission.xml [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+        android:layout_width="48dp"
+        android:layout_height="24dp"
+        android:layout_marginTop="18dp"
```
Comment (devender197) [inline app/src/main/res/layout/row_permission.xml:29]: I think, if we are using `android:gravity="center_vertical" ` for `lblPermissionName`. we should use same for `scPermission` for consistency.

Comment (devender197) [inline app/src/main/res/layout/row_permission.xml:29]: To center align view like we did for `lblPermissionName` . we can remove the top and bottom padding.

File: app/src/main/java/com/trx/consumer/models/common/PermissionModel.kt [hunk 11]
```diff
@@ -0,0 +1,23 @@
+package com.trx.consumer.models.common
+
+class PermissionModel {
```
Comment (devender197) [inline app/src/main/java/com/trx/consumer/models/common/PermissionModel.kt:3]: Permission Model on iOS and android is completely different and liked the way iOS did it [here](https://github.com/hyfn/trx-consumer-ios/pull/268/files#diff-fd500713149621f3bf515dbe2909b5956251069a2fab6ff092c1a2fadd6c5f5cR17)

File: app/src/main/java/com/trx/consumer/screens/permissions/PermissionsViewModel.kt [hunk 12]
```diff
@@ -0,0 +1,25 @@
+package com.trx.consumer.screens.permissions
+
+import androidx.hilt.lifecycle.ViewModelInject
+import com.trx.consumer.base.BaseViewModel
+import com.trx.consumer.common.CommonLiveEvent
+import com.trx.consumer.models.common.PermissionModel
+
+class PermissionsViewModel @ViewModelInject constructor() : BaseViewModel(), PermissionsListener {
```
Comment (devender197) [inline app/src/main/java/com/trx/consumer/screens/permissions/PermissionsViewModel.kt:8]: @onodude Different name is used for listener `PermissionsListener` on android and on iOS it is and  [PermissionsOptionTableViewListener](https://github.com/hyfn/trx-consumer-ios/pull/268/files#diff-96b423fa20e3d3adbe04a8c7781f74a355918a6d777390b8176ac63209f9b07bR12). and function name for listener is also different.

**Pull request closed without merging**